plugins/engines/ospray/ispc/render/utils/RandomGenerator.ispc

```diff
@@ -1128,9 +1128,16 @@ float getRandomValue(varying ScreenSample& sample, const int randomNumber)
     return rotate(randomDistribution[x][y], sample.sampleID.z % 8);
 }
 
-vec3f getRandomVector(varying ScreenSample& sample, const vec3f& normal,
-                      const int randomNumber)
+inline vec3f getRandomVector(uniform FrameBuffer* uniform fb,
+                             varying ScreenSample& sample, const vec3f& normal,
+                             const int randomNumber)
 {
+    RandomTEA rng_state;
+    varying RandomTEA* const uniform rng = &rng_state;
+    RandomTEA__Constructor(rng,
```

Comment:
It seems weird to me that you have to construct the RandomTEA object every time you get a random vector. Is this how it is done elsewhere?